### ISSUE TITLE
Use proper HTTP verbs for reward semester activation

### ISSUE THREAD

**Issue by niklasmohrin** [e-valuation/EvaP#1835]
Activating a semester for reward points is done by sending a request to `"reward_semester_activation/<int:semester_id>/<str:active>"`. Currently, this request can very well be a `GET` request. However, a `GET` request should not modify state. The idiomatic (/restful) design would be to have an endpoint to edit the `SemesterActivation` object, i.e. `/rewards/semester_activation/<int:semester_id>/edit` which must be `POST`ed to.

Afterwards, we should consider refactoring the frontend to use html forms instead of ajax.
